packages/connection/src/class-plugin-storage.php

```diff
@@ -147,7 +147,7 @@ public static function configure() {
 		}
 
 		// If a plugin was activated or deactivated.
-		$number_of_plugins_differ = count( self::$plugins ) !== count( get_option( self::ACTIVE_PLUGINS_OPTION_NAME, array() ) );
+		$number_of_plugins_differ = count( self::$plugins ) !== count( (array) get_option( self::ACTIVE_PLUGINS_OPTION_NAME, array() ) );
```

Comment:
Without being able to reproduce this, this is likely enough. Ideally, if we can run this code on an impacted site and confirm it isn't the `$plugins` that is a problematic. (I don't see how, but I don't trust myself).